src/client/tile.cpp

```diff
@@ -1811,6 +1811,23 @@ bool TextureSource::generateImagePart(std::string part_of_name,
 	return true;
 }
 
+/*
+	Calculate the color of a single pixel drawn on top of another pixel.
+
+	This is a little more complicated than just video::SColor::getInterpolated
+	because getInterpolated does not handle alpha correctly.  For example, a
+	pixel with alpha=64 drawn atop a pixel with alpha=128 should yield a
+	pixel with alpha=160, while getInterpolated would yield alpha=96.
+*/
+static inline video::SColor blitPixel(const video::SColor &src_c, const video::SColor &dst_c, u32 ratio)
+{
+	if (dst_c.getAlpha() == 0)
+		return src_c;
+	video::SColor out_c = src_c.getInterpolated(dst_c, (float)ratio / 255.0f);
+	out_c.setAlpha(dst_c.getAlpha() + (255 - dst_c.getAlpha()) * src_c.getAlpha() / 255 * ratio / 255);
```

Comment:
Are you sure that you want to multiply and divide by entire integers? Especially on low values this gets inaccurate way too fast.

Reply:
That, or convert it all into floats. i.e. `(float)ratio / 255.0f`

Reply:
Is there any specific case where doing the math in floats yields a different answer than just doing the multiplications first before the division?  Upcasting and then downcasting multiple values for each pixel seems like overkill to achieve the same final answer, and probably not great to do inside an inner loop like this.

Reply:
For this case your approach of multiplying before dividing is better. Whereas fixed/floating point casting takes negligible time, the decimals will be lost anyway.
This should work just fine:
```C++
out_c.setAlpha(dst_c.getAlpha() + (255 - dst_c.getAlpha()) * src_c.getAlpha() * ratio / (255 * 255));
```